gateway-modules/apim-ballerina-native-component/src/test/java/org/wso2/carbon/apimgt/ballerina/util/ListFilesTestCase.java

```diff
@@ -0,0 +1,160 @@
+//            } catch (IOException ioe) {
+//            }
+//        }
+
+        //      /home/sabeena
+        File notADir = new File("samples/util/notADir");
+        notADir.delete();
+
+        //boolean allJsonDirCreated = (new File("/home/sabeena/Desktop/allJsonDir")).mkdirs();
+        //boolean allJsonDirCreated = (new File("samples/util/allJsonDir")).mkdirs();
+        boolean dirCreated = (new File("samples/util/testDir")).mkdirs();
+
+        if (dirCreated) {
+            File a = new File("samples/util/testDir/a.json");
+            File b = new File("samples/util/testDir/b.txt");
+
+            try {
+                a.createNewFile();
+                b.createNewFile();
+            } catch (IOException ioe) {
+            }
+        }
+
+//        if (noJsonDirCreated) {
+//            File a = new File("samples/util/noJsonDir/a.txt");
+//            File b = new File("samples/util/noJsonDir/b.txt");
+//
+//            try {
+//                a.createNewFile();
+//                b.createNewFile();
+//            } catch (IOException ioe) {
+//            }
+//        }
+
+        bLangProgram = BTestUtils.parseBalFile("samples/util/listJsonFiles.bal");
+    }
+
+    @Test(description = "Returns a string array of json file names")
+    public void testListJsonFiles() {
+        BValue[] args = {};
+        BValue[] returns = BLangFunctions.invokeNew(bLangProgram, "listJsonFiles", args);
+
+        //Assert.assertNull(returns[0]);
+        //Assert.assertTrue(returns[0] == null);
+        Assert.assertTrue(returns[0] instanceof BStringArray);
+        Assert.assertEquals(((BStringArray) returns[0]).size(), 0);
+
+
+        Assert.assertTrue(returns[1] instanceof BStringArray);
+        BStringArray d1Files = (BStringArray) returns[1];
+        Assert.assertEquals(d1Files.size(), 1);
+
+        Assert.assertEquals(d1Files.get(0), "a.json");
+        //Assert.assertEquals(d1Files.get(1), "b.json");
+
+//        Assert.assertTrue(returns[2] instanceof BStringArray);
+//        Assert.assertEquals(((BStringArray) returns[2]).size(), 0);
+
+
+//        BStringArray d3Files = (BStringArray) returns[2];
+//        Assert.assertEquals(d3Files.size(), 2);
+//        Assert.assertEquals(d3Files.get(0), "d");
+//        Assert.assertEquals(d3Files.get(1), "e");
+
+    }
+
+    @AfterClass
+    public void deleteTestDirectories() {
+        
+        File testDir = new File("samples/util/testDir");
+        String[] jsonFiles = testDir.list();
+        for (String s : jsonFiles) {
+            File currentFile = new File(testDir.getPath(), s);
+            currentFile.delete();
+        }
+        testDir.delete();
+
+//        File noJsonDir = new File("samples/util/noJsonDir");
+//        String[] txtFiles = noJsonDir.list();
+//        for (String s : txtFiles) {
+//            File currentFile = new File(noJsonDir.getPath(), s);
+//            currentFile.delete();
+//        }
+//        noJsonDir.delete();
```

Comment:
There seems to be multiple places commented out. If there is a reason please specify it as a todo. Otherwise shall we clean them>

Reply:
okay